#! /bin/sh -e 
## ocamlobjinfo.dpatch by <luther@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: ship ocamlobjinfo and add a manpage for it.

if [ $# -ne 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;	
    *)
	echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
	exit 1;;
esac

exit 0
@DPATCH@
diff -urNad ocaml~/man/ocamlc.m ocaml/man/ocamlc.m
--- ocaml~/man/ocamlc.m	2008-11-29 11:16:39.000000000 +0100
+++ ocaml/man/ocamlc.m	2008-11-29 11:31:06.000000000 +0100
@@ -579,7 +579,7 @@
 Display a short usage summary and exit.
 
 .SH SEE ALSO
-.BR ocamlopt (1), \ ocamlrun (1), \ ocaml (1).
+.BR ocamlopt (1), \ ocamlrun (1), \ ocaml (1), \ ocamlobjinfo (1).
 .br
 .IR "The Objective Caml user's manual" ,
 chapter "Batch compilation".
diff -urNad ocaml~/man/ocamlobjinfo.m ocaml/man/ocamlobjinfo.m
--- ocaml~/man/ocamlobjinfo.m	1970-01-01 01:00:00.000000000 +0100
+++ ocaml/man/ocamlobjinfo.m	2008-11-29 11:31:10.000000000 +0100
@@ -0,0 +1,40 @@
+.TH OCAML-SOAP 1 "October 26, 2002"
+.SH NAME
+ocamlobjinfo \- dump an OCaml compilation unit description
+.SH SYNOPSIS
+.B ocamlobjinfo
+.RI file\ ...
+.SH DESCRIPTION
+Dump information contained in OCaml compilation units.
+Currently it works on .cmi, .cmo and .cma files.
+.sp 2
+.B ocamlobjinfo
+is able to show information regarding:
+.br
+.IP
+module names
+.sp 2
+unit name
+.sp 2
+declared primitives
+.sp 2
+imported interfaces
+.sp 2
+md5sums of imported interfaces
+.sp 2
+forced custom mode
+.sp 2
+extra C libraries needed
+.sp 2
+extra C flags needed
+.sp 2
+use of unsafe features
+.PP
+depending on its invocation on .cmi, .cmo or .cma arguments.
+.SH SEE ALSO
+.BR ocamlc (1),
+.BR ocamlopt (1)
+.br
+.SH AUTHOR
+This manual page was written by Stefano Zacchiroli <zack@debian.org>,
+for the Debian GNU/Linux system (but may be used by others).
diff -urNad ocaml~/man/ocamlopt.m ocaml/man/ocamlopt.m
--- ocaml~/man/ocamlopt.m	2008-11-29 11:16:39.000000000 +0100
+++ ocaml/man/ocamlopt.m	2008-11-29 11:31:10.000000000 +0100
@@ -603,7 +603,8 @@
 SPARC processors.
 
 .SH SEE ALSO
-.BR ocamlc (1).
+.BR ocamlc (1),
+.BR ocamlobjinfo (1).
 .br
 .IR "The Objective Caml user's manual" ,
 chapter "Native-code compilation".
diff -urNad ocaml~/tools/Makefile.shared ocaml/tools/Makefile.shared
--- ocaml~/tools/Makefile.shared	2007-11-22 23:14:43.000000000 +0100
+++ ocaml/tools/Makefile.shared	2008-11-29 11:31:10.000000000 +0100
@@ -24,7 +24,7 @@
 LINKFLAGS=$(INCLUDES)
 
 all: ocamldep ocamlprof ocamlcp ocamlmktop ocamlmklib scrapelabels addlabels \
-     dumpobj
+     dumpobj objinfo
 .PHONY: all
 
 opt.opt: ocamldep.opt
@@ -72,6 +72,7 @@
 	cp ocamlprof $(BINDIR)/ocamlprof$(EXE)
 	cp ocamlcp $(BINDIR)/ocamlcp$(EXE)
 	cp profiling.cmi profiling.cmo $(LIBDIR)
+	cp objinfo $(BINDIR)/ocamlobjinfo$(EXE)
 
 clean::
 	rm -f ocamlprof ocamlcp
